library/core/src/num/mod.rs

```diff
@@ -846,37 +866,42 @@ fn from_str_radix<T: FromStrRadixHelper>(src: &str, radix: u32) -> Result<T, Par
     };
 
     let mut result = T::from_u32(0);
-    if is_positive {
-        // The number is positive
-        for &c in digits {
-            let x = match (c as char).to_digit(radix) {
-                Some(x) => x,
-                None => return Err(PIE { kind: InvalidDigit }),
-            };
-            result = match result.checked_mul(radix) {
-                Some(result) => result,
-                None => return Err(PIE { kind: PosOverflow }),
-            };
-            result = match result.checked_add(x) {
-                Some(result) => result,
-                None => return Err(PIE { kind: PosOverflow }),
-            };
+
+    if intrinsics::likely(
```

Comment:
Adding a fastpath for the common case is one thing, but `likely` and `unlikely` can make the else branch a lot slower for minor gains on the if branch. Unless you're absolutely sure that in your workload one branch is taken in the overwhelming majority of cases, I'd leave it up to the compiler. Since this is std, you really don't know much about the workload and may very well be slowing down people parsing numbers `>= 10,000,000` to an `i32`. Or even just numbers `>= 10` to an `i8`.

Reply:
Law of unintended consequences! Ok will whip that off.